src/Silicon.cpp

```diff
@@ -959,39 +1287,210 @@ namespace galsim {
                 dbg<<"ix,iy = "<<ix<<','<<iy<<"  x,y = "<<x<<','<<y<<std::endl;
                 set_verbose(2);
                 bool off_edge;
-                insidePixel(ix, iy, x, y, zconv, target, &off_edge);
-                searchNeighbors(*this, ix, iy, x, y, zconv, target, step);
+                insidePixel(ix, iy, x, y, zconv, target, &off_edge,
+                            emptypolySize, pixelInnerBoundsData,
+                            pixelOuterBoundsData,
+                            horizontalBoundaryPointsData,
+                            verticalBoundaryPointsData,
+                            emptypolyData);
+                searchNeighbors(*this, ix, iy, x, y, zconv, b, step, emptyPolySize,
+                                pixelInnerBoundsData, pixelOuterBoundsData,
+                                horizontalBoundaryPointsData,
+                                verticalBoundaryPointsData, emptyPolyData);
                 set_verbose(1);
 #endif
-                int n = (pixelNotFoundRandom[i-i1] > 0.5) ? 0 : step;
+                const int xoff[9] = {0,1,1,0,-1,-1,-1,0,1}; // Displacements to neighboring pixels
+                const int yoff[9] = {0,0,1,1,1,0,-1,-1,-1}; // Displacements to neighboring pixels
+                // use uniform random numbers for pixel not found from randomArray
+                // (3rd of 4 numbers for this photon)
+                int n = (randomArray[(i-i1) * 4 + 2] > 0.5) ? 0 : step;
                 ix = ix + xoff[n];
                 iy = iy + yoff[n];
             }
 
-            if (b.includes(ix,iy)) {
+            if (b.includes(ix, iy)) {
+                int deltaIdx = (ix - deltaXMin) * deltaStep + (iy - deltaYMin) * deltaStride;
 #ifdef _OPENMP
 #pragma omp atomic
 #endif
-                _delta(ix,iy) += flux;
+                deltaData[deltaIdx] += flux;
 
                 // This isn't atomic -- openmp is handling the reduction for us.
                 addedFlux += flux;
             }
         }
-#ifdef _OPENMP
-        }
-#endif
+        
         return addedFlux;
     }
 
+    void Silicon::updatePixelBounds(int nx, int ny, size_t k,
+                                    Bounds<double>* pixelInnerBoundsData,
+                                    Bounds<double>* pixelOuterBoundsData,
+                                    Position<float>* horizontalBoundaryPointsData,
+                                    Position<float>* verticalBoundaryPointsData)
```

Comment:
Please move this up to where this function used to be so it shows up as a diff, rather than an whole new function (and a deleted one above).